homework/src/index.js

```diff
@@ -22,25 +23,114 @@
     Object.keys(options).forEach((key) => {
       const value = options[key];
       if (key === 'text') {
-        elem.innerText = value;
+        elem.innerHTML = value;
       } else {
         elem.setAttribute(key, value);
       }
     });
     return elem;
   }
 
+  const root = document.getElementById('root');
+  const div = createAndAppend('div', root, { class: 'data' });
+  const header = createAndAppend('p', div, { text: 'HYF Repositories', class: 'header' });
+  const select = createAndAppend('select', header, { id: 'list' });
+  const table = createAndAppend('table', root, { class: 'table' });
+  const tbody = createAndAppend('tbody', table);
+  const tr = createAndAppend('tr', tbody, { text: 'Repository: ', class: 'label' });
+
+  const repositoryLink = createAndAppend('a', tr);
+  const description = createAndAppend('tr', tbody, { class: 'label' });
+  const forks = createAndAppend('tr', tbody, { class: 'label' });
+  const container = createAndAppend('div', root, { id: 'container' });
+  const updated = createAndAppend('tr', tbody, { class: 'label' });
+
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
-    });
-  }
+    fetchJSON(url)
+      .then(repositories => {
+        repositories.forEach((item, index) => {
+          createAndAppend('option', select, {
+            text: item.name,
+            value: index
+          });
+        });
 
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
+        select.addEventListener('change', () => {
+          container.innerHTML = "";
+          const index = event.target.value;
+
+          renderRepositoryBox(
+            repositories[select.value],
+          );
+          const contributors_url = repositories[index].contributors_url;
+          fetchJSON(contributors_url)
+            .then(contributorData => {
+              renderContributors(contributorData, container);
+            })
+            .catch(error => {
+              createAndAppend('div', container, {
+                text: error.message,
+                class: 'alert-error'
+              });
+              return;
+            });
+
+        });
+
+        const contributorsInfo = repositories[0].contributors_url;
+        fetchJSON(contributorsInfo)
+          .then(contributorData => {
+            renderContributors(contributorData, container);
+          })
+          .catch(error => {
+            createAndAppend('div', container, {
+              text: error.message,
+              class: 'alert-error'
+            });
+            return;
+          });
+
+        function renderRepositoryBox(repositories) {
```

Comment:
@J05J4 made this comment in the week 1 review:

>You have to improve your code to make it clean: there are functions inside of functions and there are function declarations underneath the call to that function. You need to update this (could also be in your week two solution). If you do not understand this completely please ask questions in the next lesson, because this is really important!

I expected that you made the necessary changes in your week 2 homework, but I don't see them.

Reply:
Hello Jim, Thanks for your feedback, Now i understand that functions should not ne in another function, and what should i menthioned in Main function. I have already made changes in JS-WEEK3 and splited functions from each other to be moe redable. Tomorrow i will make a pull request for JS_WEEK3.  And after that i will change week2 and week 1 as soon as possible :)